commit/plugin_e2e_test.go

```diff
@@ -417,6 +416,208 @@ func TestPlugin_E2E_AllNodesAgree_TokenPrices(t *testing.T) {
+	params := defaultNodeParams(t)
+	merkleOutcome := baseMerkleOutcome(params.rmnReportCfg)
+	nodes := make([]ocr3types.ReportingPlugin[[]byte], len(oracleIDs))
+
+	newFeeComponents, newNativePrice, packedGasPrice := newRandomFees()
+	expectedChainFeeOutcome := chainfee.Outcome{
+		GasPrices: []ccipocr3.GasPriceChain{
+			{
+				GasPrice: packedGasPrice,
+				ChainSel: destChain,
+			},
+		},
+	}
+
+	newFeeComponents2, newNativePrice2, newPackedGasPrice2 := newRandomFees()
+
+	testCases := []struct {
+		name                  string
+		prevOutcome           Outcome
+		expOutcome            Outcome
+		expTransmittedReports []ccipocr3.CommitPluginReport
+
+		mockCCIPReader func(*readerpkg_mock.MockCCIPReader)
+	}{
+		{
+			name:        "fee components should be updated",
+			prevOutcome: Outcome{},
+			expOutcome: Outcome{
+				MerkleRootOutcome: merkleOutcome,
+				ChainFeeOutcome: chainfee.Outcome{
+					GasPrices: []ccipocr3.GasPriceChain{
+						{
+							GasPrice: packedGasPrice,
+							ChainSel: destChain,
+						},
+						{
+							GasPrice: packedGasPrice,
+							ChainSel: sourceChain1,
+						},
+						{
+							GasPrice: packedGasPrice,
+							ChainSel: sourceChain2,
+						},
+					},
+				},
+			},
+			mockCCIPReader: func(m *readerpkg_mock.MockCCIPReader) {
+				m.EXPECT().GetAvailableChainsFeeComponents(params.ctx, mock.Anything).Unset()
+				m.EXPECT().
+					GetAvailableChainsFeeComponents(params.ctx, mock.Anything).
+					Return(
+						map[ccipocr3.ChainSelector]types.ChainFeeComponents{
+							destChain:    newFeeComponents,
+							sourceChain1: newFeeComponents,
+							sourceChain2: newFeeComponents,
+						}).
+					Maybe()
+
+				m.EXPECT().GetWrappedNativeTokenPriceUSD(params.ctx, mock.Anything).Unset()
+				m.EXPECT().
+					GetWrappedNativeTokenPriceUSD(params.ctx, mock.Anything).
+					Return(map[ccipocr3.ChainSelector]ccipocr3.BigInt{
+						destChain:    newNativePrice,
+						sourceChain1: newNativePrice,
+						sourceChain2: newNativePrice,
+					}).Maybe()
+			},
+		},
+		{
+			name:        "fee components should be updated when there's a subset of chains",
+			prevOutcome: Outcome{},
+			expOutcome: Outcome{
+				MerkleRootOutcome: merkleOutcome,
+				ChainFeeOutcome:   expectedChainFeeOutcome,
+			},
+			mockCCIPReader: func(m *readerpkg_mock.MockCCIPReader) {
+				m.EXPECT().GetAvailableChainsFeeComponents(params.ctx, mock.Anything).Unset()
+				m.EXPECT().
+					GetAvailableChainsFeeComponents(params.ctx, mock.Anything).
+					Return(
+						map[ccipocr3.ChainSelector]types.ChainFeeComponents{
+							destChain: newFeeComponents,
+						}).
+					Maybe()
+
+				m.EXPECT().GetWrappedNativeTokenPriceUSD(params.ctx, mock.Anything).Unset()
+				m.EXPECT().
+					GetWrappedNativeTokenPriceUSD(params.ctx, mock.Anything).
+					Return(map[ccipocr3.ChainSelector]ccipocr3.BigInt{
+						destChain:    newNativePrice,
+						sourceChain1: newNativePrice,
+						sourceChain2: newNativePrice,
```

Comment:
The assertions would be a lot easier to read if the parameter was these prices rather than the prices and all the expectations